casual/casual-jca/src/main/java/se/laz/casual/jca/CasualResourceAdapter.java

```diff
@@ -219,10 +225,15 @@ private void startWork(Work work, WorkListener workListener)
     public void endpointDeactivation(MessageEndpointFactory endpointFactory,
                                      ActivationSpec spec)
     {
-        log.finest(()->"endpointDeactivation()");
+        log.info(()->"endpointDeactivation() ");
+        RuntimeInformation.setDomainIsBeingShutdown(true);
         InboundDeactivatedContext.domainDisconnect();
         InboundDeactivatedContext.clear();
         InboundTopologyUpdateContext.clear();
+        Predicate predicate = () -> inboundTransactionRegistry.hasPending() || CasualResourceManager.getInstance().hasPending();
```

Comment:
What is the difference here, would `inboundTransactionRegistry.hasPending()`, not always be a the same as `CasualResourceManager.getInstance().hasPending()`?